### PR TITLE
{devel}[intelcuda/2017b] PyTorch v0.3.1

### PR DESCRIPTION
**Pull request by vanzod**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by vanzod** [comment]
I tried again to get it to build with `pip` but the problem is that the `cmake` invoked by `setyp.py` fails to find the MKL library and the test fails with:
```
======================================================================
ERROR: test_DoubleTensor_inverse (__main__.TestCuda)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test_cuda.py", line 412, in tmp
    cpu_result = getattr(cpu_tensor, fn)(*cpu_args)
RuntimeError: getrf : Lapack library not found in compile time
 at /tmp/eb-a4oGxp/pip-9ARwjr-build/torch/lib/TH/generic/THLapack.c:140
```
This does not happen when the installation is done by invoking `setup.py` directly.
All attempts to force CMake to pick up MKL failed for me.

@boegel Do you want to whitelist it for now?

**Comment by boegel** [comment]
@vanzod whitelisting for `PyTorch` & `use_pip` done in #7844

**Comment by vanzod** [comment]
Re-trigger Travis

**Comment by vanzod** [comment]
Re-trigger Travis

**Comment by wpoely86** [comment]
@vanzod you still need to add `download_dep_fail` to the pytorch easyconfigs

**Comment by vanzod** [comment]
Test report by @vanzod
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in this PR)
cermis - Linux debian 9.8, Intel(R) Xeon(R) CPU           X5670  @ 2.93GHz, Python 2.7.13
See https://gist.github.com/243a6b539818c8af932c8262989cdf2f for a full test report.

File: easybuild/easyconfigs/p/PyTorch/PyTorch-0.3.1-intelcuda-2017b-Python-2.7.14.eb [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    '710bbd38973a94c5c20695d25b819c4b41565bd4723ee4e07542d9537fa2bf18',  # PyTorch-0.3.1-cmake.patch
+    '72c8d6804bff568c2a49b1b7019171c93309a1853a90d5e3abe778f8aa80ca4f',  # PyTorch-0.3.1_skip_MPI_tests.patch
+    '2c07d9c3622986df3ec8d7f761dc6984f1b44c7a377cad7d7667a030aaeb526f',  # PyTorch-0.3.1_intel_skip_autgrad_test.patch
+    '951d5fde89df3be98c1bc6750546259c1b3d0c77574fa3f9e67d9e8fe5b08161',  # PyTorch-0.3.1_fix_findAVX.patch
+    '400c2e5c2fc4ec0b03d8b95c4612fa174f924e65c23e83b50e21556a36d8f293',  # PyTorch-0.3.1_Fix_pushback_type.patch
+    'b0a6dc564aff1cd6aba0c0c77175413c8f8dc03e771d374755df48365d84bb74',  # PyTorch-0.3.1_Fix_static_assert.patch
+]
+
+# PyTorch pulls in a bunch of submodules which don't have releases.
+# We download the submodule revisions from their repos.
```
Comment (akesandgren) [inline easybuild/easyconfigs/p/PyTorch/PyTorch-0.3.1-intelcuda-2017b-Python-2.7.14.eb]: These comments are probably in the wrong place... used --inject-checksum perhaps?

File: easybuild/easyconfigs/p/PyTorch/PyTorch-0.3.1-intelcuda-2017b-Python-3.6.3.eb [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    '710bbd38973a94c5c20695d25b819c4b41565bd4723ee4e07542d9537fa2bf18',  # PyTorch-0.3.1-cmake.patch
+    '72c8d6804bff568c2a49b1b7019171c93309a1853a90d5e3abe778f8aa80ca4f',  # PyTorch-0.3.1_skip_MPI_tests.patch
+    '2c07d9c3622986df3ec8d7f761dc6984f1b44c7a377cad7d7667a030aaeb526f',  # PyTorch-0.3.1_intel_skip_autgrad_test.patch
+    '951d5fde89df3be98c1bc6750546259c1b3d0c77574fa3f9e67d9e8fe5b08161',  # PyTorch-0.3.1_fix_findAVX.patch
+    '400c2e5c2fc4ec0b03d8b95c4612fa174f924e65c23e83b50e21556a36d8f293',  # PyTorch-0.3.1_Fix_pushback_type.patch
+    'b0a6dc564aff1cd6aba0c0c77175413c8f8dc03e771d374755df48365d84bb74',  # PyTorch-0.3.1_Fix_static_assert.patch
+]
+
+# PyTorch pulls in a bunch of submodules which don't have releases.
+# We download the submodule revisions from their repos.
```
Comment (akesandgren) [inline easybuild/easyconfigs/p/PyTorch/PyTorch-0.3.1-intelcuda-2017b-Python-3.6.3.eb]: Same thing here...

File: easybuild/easyconfigs/p/PyTorch/PyTorch-0.3.1_Fix_static_assert.patch [hunk 3]
```diff
@@ -0,0 +1,32 @@
+Replace designated initializers that are not part of standard C++ and
+upset the Intel 2017 compiler.
```
Comment (akesandgren) [inline easybuild/easyconfigs/p/PyTorch/PyTorch-0.3.1_Fix_static_assert.patch]: This text looks wrong, deginated initializer is the other patch...

**Comment by akesandgren** [comment]
Failing PyTorch runtest due to
ImportError: /usr/lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.22' not found (required by /scratch/eb-ake/PyTorch/0.3.1/intelcuda-2017b-Python-2.7.14/pytorch-0.3.1/build/lib.linux-x86_64-2.7/torch/_C.so)

Isn't the correct modules loaded when runtest is executed?

**Comment by akesandgren** [comment]
Test report by @akesandgren
**FAILED**
Build succeeded for 2 out of 4 (4 easyconfigs in this PR)
b-an03.hpc2n.umu.se - Linux ubuntu 16.04, Intel(R) Xeon(R) CPU E5-2690 v4 @ 2.60GHz, Python 2.7.12
See https://gist.github.com/cba3396db0f622d728c07ae7224e9fb6 for a full test report.

**Comment by akesandgren** [comment]
@vanzod ping

**Comment by vanzod** [comment]
The toolchain is now quite old. Closing PR

**Pull request closed without merging**